src/components/ConfirmPopup.js

```diff
@@ -0,0 +1,34 @@
+import Popup from "./Popup.js";
+
+export default class ConfirmPopup extends Popup {
+  constructor({ popupSelector }) {
+    super({ popupSelector });
+    this._submitFunction = null;
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._saveButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setSubmitAction(submitFunction) {
+    this._submitFunction = submitFunction;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._saveButton.textContent = "Loading...";
+    } else {
+      this._saveButton.textContent = "Yes";
+    }
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+
+    this._submitButton.addEventListener("click", (e) => {
+      e.preventDefault();
+      if (this._submitFunction) {
+        this._submitFunction();
+      }
+      this.close();
```

Comment:
Please, delete it

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

Reply:
I am confused, I thought I was already doing this, for example here: function handleEditAvatarSubmit(formData) {
  editAvatarPopup.renderLoading(true);
  const avatarUrl = formData.avatarUrl;
  api
    .updateProPic(avatarUrl)
    .then((updatedUserData) => {
      console.log(updatedUserData);
      userInfo.setuserAvatar(updatedUserData.avatar);
      editAvatarPopup.close();
    })
    .catch((error) => {
      console.error("Error updating avatar:", error);
    })
    .finally(() => {
      editAvatarPopup.renderLoading(false, "Save");
    });
} close call back is within then block already. I will submit this again and hopefully you can clarify further, thank you!

Reply:
> close call back is within then block already. I will submit this again and hopefully you can clarify further, thank you!

I meant that you need to delete line 31 from `ConfirmPopup.js`. You shouldn't close the modal there